client/src/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import $ from 'jquery';
+import styled from 'styled-components';
+import NameGenderCategoryPrice from './components/NameGenderCategoryPrice.jsx';
+import Styles from './components/Styles.jsx';
+
+const ProductNeedToKnow = styled.div`
+  display: flex;
+  flex-flow: column;
+`;
+
+class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      productId: '',
+      name: '',
+      gender: '',
+      category: '',
+      style: {},
+      styles: [],
+    };
+  }
+
+  componentDidMount() {
+    console.log('fetching...');
+    $.ajax({
+      url: '/products/all/1'
+    }).done((data) => {
+      console.log('server', data);
+      let d = data[0];
+      this.setState({
+        productId: d.productId,
+        name: d.name,
+        gender: d.gender === 'male' ? 'Men\'s' : 'Women\'s',
+        category: d.category,
+        style: d.styles[0],
+        styles: d.styles,
+      });
+      console.log('state', this.state);
+    });
+  }
+
+  render() {
+    console.log(this.state);
+    return (
+      <ProductNeedToKnow>
```

Comment:
I'm not sure what the name of this component is supposed to mean. Would a more functional name like "ProductInfo" be more clear?

Full disclosure: I don't know styled components so perhaps this makes sense in that architecture. Same disclaimer goes for all the comments below

Reply:
I prefer to use `need to know` as this is product information that a customer needs to know in order to make a purchase.

ProductInfo is general and may have tons of other information like product description, etc.

Reply:
On second thought, I revised this to ProductInfo 😄

Reply:
Your reasoning makes sense as well! Haha